### PR TITLE
TST: Raise ValueError and suggestion to use header=None if header=-1 is pa…

### LINKED ISSUE #27779: Give more precise warning when passing `header=-1` in `pandas.read_csv`, or don't throw error
#### Code Sample, a copy-pastable example if possible

```python
from io import StringIO
import pandas as pd

print(f"Using pandas version {pd.__version__}")
pd.read_csv(StringIO('spam'), header=-1)
``` 
> Using pandas version 0.25.0
>
>---------------------------------------------------------------------------
>OverflowError  
#### Problem description

In previous versions of pandas (e.g. 0.24.2), this would have worked fine. Instead, now, one has to pass `header=None`. However, when working with old code after having updated pandas, it may be unclear where the problem lies.

#### Expected Output
```
      0
0  spam
```

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.3.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-55-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 0.25.0
numpy            : 1.16.4
pytz             : 2019.1
dateutil         : 2.8.0
pip              : 19.1.1
setuptools       : 41.0.1
Cython           : 0.29.12
pytest           : 4.5.0
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.7.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.0
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.13.0
pytables         : None
s3fs             : 0.2.1
scipy            : 1.3.0
sqlalchemy       : 1.3.5
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by MarcoGorelli**
…ssed

- [x] closes #27779
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.25.1.rst [hunk 1]
```diff
@@ -105,7 +105,7 @@ I/O
 ^^^
 
 - Avoid calling ``S3File.s3`` when reading parquet, as this was removed in s3fs version 0.3.0 (:issue:`27756`)
--
+- Advise user to use ``header=None`` rather than ``header=-1`` for no header in ``pandas.read_csv`` (:issue:`27779`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.25.1.rst]:
```suggestion
- Better error message when a negative header is passed in :func:`pandas.read_csv` (:issue:`27779`)
```

File: pandas/tests/io/parser/test_header.py [hunk 2]
```diff
@@ -24,6 +24,20 @@ def test_read_with_bad_header(all_parsers):
         parser.read_csv(s, header=[10])
 
 
+def test_negative_header(all_parsers):
+    parser = all_parsers
```
Comment (TomAugspurger) [inline pandas/tests/io/parser/test_header.py]: Can you add the issue number as a comment here?

File: pandas/io/common.py [hunk 3]
```diff
@@ -125,6 +125,12 @@ def _validate_header_arg(header) -> None:
             "the row(s) making up the column names"
         )
 
+    # In version 0.24, this would have worked.
+    if header == -1:
```
Comment (jreback) [inline pandas/io/common.py]: this should only validate if header is an int (or it could be an array-like of ints); these should raise if < 0 right?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/common.py (hunk 3 above)
Comment (gfyoung) [reply pandas/io/common.py]: @jreback : I agree that this check needs to be expanded as you described

Comment (gfyoung) [reply pandas/io/common.py]: The test that was added will of course need to be updated as well

**Review by gfyoung — COMMENTED** [review]
@MarcoGorelli : This PR is off to a good start!  If you can address @jreback's comment and expand the test, I think this will address the issue well.

File: pandas/io/common.py (hunk 3 above)
Comment (MarcoGorelli) [reply pandas/io/common.py]: Sure - thank you all for having taken the time to review my work!

**Comment by TomAugspurger** [comment]
Thanks @MarcoGorelli!

**Comment by MarcoGorelli** [comment]
Happy to help, and massive thanks for your assistance/guidance!

**Pull request merged**